ccm_web/server/src/api/api.service.ts

```diff
@@ -124,6 +134,56 @@ export class APIService {
     return await sectionHandler.enrollUsers(sectionUsers)
   }
 
+  async createExternalUsers (user: User, externalUsers: ExternalUserDto[]): Promise<ExternalUserCreationResult> {
+    const adminRequestor = this.canvasService.createRequestorForAdmin('/api/v1/')
+    const adminHandler = new AdminApiHandler(adminRequestor, user.loginId)
+    const newUserAccountID = this.configService.get('canvas.newUserAccountID', { infer: true })
+
+    const resultData: ExternalUserData = {}
+    const createErrors: APIErrorData[] = []
+    const newUsers: CanvasUserLoginEmail[] = []
+    const createUserResponses = await adminHandler.createExternalUsers(externalUsers, newUserAccountID)
+
+    // Handle create user responses: success, failure, and already exists
+    createUserResponses.forEach(({ email, result }) => {
+      let userCreated: false | APIErrorData | CanvasUserLoginEmail
+      if (isAPIErrorData(result)) {
+        if (result.statusCode === HttpStatus.BAD_REQUEST) {
+          userCreated = false
```

Comment:
I feel a little uncomfortable about assuming all `BAD_REQUEST` responses are the case that they already exist, but I'm not sure what else to do here (other than maybe parsing the response message). I guess if we feel confident in the setup of the request, responses with this code should always be this case, but I still feel a little nervous. Any thoughts on this? Not asking for anything to be done yet.

Reply:
See my PR https://github.com/lsloan/ccm/pull/7

Reply:
Looks good!  I suggested a very minor change.  You can take the suggestion or reject it if you think it's unnecessary.  Then I'll merge it.